flake.nix

```diff
@@ -0,0 +1,68 @@
+{
+  description = "ogmios-datum-cache";
+
+  inputs = {
+    haskell-nix = {
+      url = "github:input-output-hk/haskell.nix";
+      inputs.nixpkgs.follows = "haskell-nix/nixpkgs-2111";
+    };
+    nixpkgs.follows = "haskell-nix/nixpkgs-unstable";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+
+      perSystem = nixpkgs.lib.genAttrs supportedSystems;
+
+      nixpkgsFor = system: import nixpkgs { inherit system; overlays = [ haskell-nix.overlay ]; inherit (haskell-nix) config; };
+
+      projectFor = system:
+        let
+          deferPluginErrors = true;
+          pkgs = nixpkgsFor system;
+
+          fakeSrc = pkgs.runCommand "real-source" {} ''
```

Comment:
Related to my comment above about using the MLabs fork: you can see an example of using the new way to use `extraSources` here: https://github.com/Plutonomicon/cardano-transaction-lib/blob/6fc892b4577ad71c9d20f82d4dd8b7fdc2f559f6/server/nix/default.nix#L9